one_fm/jinja/print_format/methods.py

```diff
@@ -0,0 +1,131 @@
+import frappe
+from datetime import date, datetime
+from one_fm.utils import _dateutil
+
+class PrintFormat:
+    """
+    Print format class
+    """
+
+    def sic_attendance_absent_present(self, doc):
+        """
+        Print format for absent/present in sales invoice
+        for Contracts
+        """
+        # print format
+        template, context = sic_attendance_absent_present(doc)
+        return frappe.render_template(
+            template, context
+        )
+
+pf = PrintFormat()
+
+
+# ATTENDANCE MAPS
+attendance_map = {
```

Comment:
We already have abbreviations for attendance map in the roster something like below:

attendance_abbr_map = {
  'Present': 'P',
  'Absent': 'A',
  'Work From Home': 'WFH',
  'Half Day': 'HD',
  'On Leave': 'OL'
};

I recommend using that.

Reply:
The PF have specific map.

Reply:
We could go with anything tbh. Let's stick with what is already implemented in the application.
Just provide a legend in the print format for the attendance map.